### PR TITLE
Feature - extended CS sniffs

### PR DESCRIPTION
**Pull request by webimpress**
Please see: https://github.com/zendframework/zend-coding-standard/pull/1

### CODE REVIEW

**Comment by webimpress** [comment]
In this PR I've included some changes, probably it should be separate: classes `ClassGenerator` and `TraitUsageGenerator` now implements interface `TraitUsageInterface`. This interface has not been implemented anywhere but doc comments in these classes pointed to this interface. I think it is a bug in the library and it should be fixed in separate PR. I've did it here on purposes of this PR (for now it is just to show changes required by extended CS sniffs from zend-coding-standard library).

@Ocramius is there any other purpose of this not used interface, or I am right, and it should be used in these two classes?

File: src/Generator/ClassGenerator.php [hunk 1]
```diff
@@ -11,33 +11,33 @@
 
 use Zend\Code\Reflection\ClassReflection;
 
-class ClassGenerator extends AbstractGenerator
+class ClassGenerator extends AbstractGenerator implements TraitUsageInterface
```
Comment (Ocramius) [inline src/Generator/ClassGenerator.php]: Uhhh, unsure why this class should ever need `TraitUsageInterface` :-\

Comment (webimpress) [inline src/Generator/ClassGenerator.php]: @Ocramius Please check out my comment https://github.com/zendframework/zend-code/pull/109#issuecomment-307087267
Please note that this class __has all methods__ from this interface and in doc comments of these methods we had:
`@inherit Zend\Code\Generator\TraitUsageInterface`
what I've changed to `@inheritDoc` (but this is fine only when class has 'parent class' or 'interface' with these methods)

File: phpcs.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    <rule ref="ZendCodingStandard.Commenting.FunctionComment.TypeHintMissing">
+        <severity>0</severity>
+    </rule>
+
+    <rule ref="ZendCodingStandard.NamingConventions.ValidVariableName.NotCamelCaps">
+        <exclude-pattern>src/Scanner/AnnotationScanner.php</exclude-pattern>
+        <exclude-pattern>src/Scanner/ClassScanner.php</exclude-pattern>
+        <exclude-pattern>src/Scanner/DocBlockScanner.php</exclude-pattern>
+        <exclude-pattern>src/Scanner/MethodScanner.php</exclude-pattern>
+        <exclude-pattern>src/Scanner/TokenArrayScanner.php</exclude-pattern>
```
Comment (webimpress) [inline phpcs.xml]: In these files we have variables like: `$MACRO_STREAM_ADVANCE_CHAR`. Probably we can rename all these variables there and then remove this exception here. Or is there any other reason why we have uppercase, underscore separated variable names there?

**Comment by Ocramius** [comment]
I manually merged this patch to `develop` via zendframework/zend-code@1dd02307a99f463f637deab1dd11a8f6e5fa004e, thanks @webimpress!

**Pull request closed without merging**